fix(linter): resolve configured rules for every file linted by `tsgolint`

camchenry opened the pull request:
The previous implementation of config resolution was simply incorrect. It did not resolve rule severities and overrides and nested configs properly. Now, although inefficient, we resolve the config file for every path that we will lint with `tsgolint` to correctly account for any possible nested configs, overrides, and so on.

camchenry commented:
* **#12886** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12886?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12886?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=12886).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
I don't understand config resolution anymore, @camc314 please review carefully.

Sysix commented on apps/oxlint/src/lint.rs:
```diff
@@ -1215,6 +1215,8 @@ mod test {
 
     #[test]
+    // FIXME: Skip on linux arm64 for now, since we are not running tsgolint there correctly it seems
+    #[cfg(all(not(target_os = "linux"), not(target_arch = "aarch64")))]
```
All test cases I could think of:
- With config flag (so base config is triggered)
- With nested config (`.oxlintrc.json` + file should be in a subdirectory)
- With `overrrides`, you already got it
- With `extends`

camc314 commented on apps/oxlint/src/tsgolint.rs:
```diff
@@ -131,19 +132,18 @@ impl<'a> TsGoLintState<'a> {
-                                }
-                            })
+                        let Some(resolved_config) = resolved_configs.get(&path) else {
+                            // If we don't have a resolved config for this path, skip it. We should always
+                            // have a resolved config though, since we processed them already above.
```
lets add a debug assertion?

camc314 reviewed (approved):
@camchenry this looks great, do you mind adding the extra tests @Sysix mentioned in a follow up PR?

camchenry committed:
fix(linter): resolve configured rules for every file linted by `tsgolint` (#12886)

The previous implementation of config resolution was simply incorrect. It did not resolve rule severities and overrides and nested configs properly. Now, although inefficient, we resolve the config file for every path that we will lint with `tsgolint` to correctly account for any possible nested configs, overrides, and so on.
